pkg/apis/apps/validation/validation.go

```diff
@@ -94,6 +97,25 @@ func ValidatePersistentVolumeClaimRetentionPolicy(policy *apps.StatefulSetPersis
 	return allErrs
 }
 
+func volumesToAddForTemplates(spec *apps.StatefulSetSpec) []api.Volume {
+	if len(spec.VolumeClaimTemplates) == 0 {
+		return nil
+	}
+	newVolumes := make([]api.Volume, 0, len(spec.VolumeClaimTemplates))
+	for i := range spec.VolumeClaimTemplates {
+		claimName := spec.VolumeClaimTemplates[i].Name
+		newVolumes = append(newVolumes, api.Volume{
```

Comment:
This will incorrectly return volumes that overlap with already-present volumes in the pod spec

If the pod already defines volumes a,b,c, and the volumeClaimTemplates define c,d,e, this should just return []{d,e}. Otherwise, when we append, `c` will be duplicated and fail validation.

Add a unit test checking that scenario.

Reply:
done